API/NoDocs/Microsoft.Extensions.Http.Telemetry.Latency/HttpClientLatencyTelemetryExtensions.cs

```diff
@@ -0,0 +1,16 @@
+// Assembly 'Microsoft.Extensions.Http.Telemetry'
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Http.Telemetry.Logging;
+
+namespace Microsoft.Extensions.Http.Telemetry.Latency;
+
+public static class HttpClientLatencyTelemetryExtensions
+{
+    public static IServiceCollection AddDefaultHttpClientLatencyTelemetry(this IServiceCollection services);
```

Comment:
yes plaese!

Reply:
haha, I actually just talked with Martin earlier and we decided not to do that particular change. I think it would only make sense to put this method on IHttpClientBuilder if we were also doing the work to make the feature isolated to particular HttpClients. If we only renamed and retargetted the method to IHttpClientBuilder without any deeper implementation changes it would still affect all HttpClients regardless of which builder it was invoked with. That would be very confusing. Adding this on IServiceCollection with container-wide effect now doesn't preclude us from adding a feature in the future that isolates it to a specific client.

Reply:
I really dislike adding things to the service collection that are targeted for sub builders. It makes naming more difficult and it always shows up, even when the top level thing doesn't apply in all situations:

```C#
services.AddLogging(b => b.AddConsole());
```

vs

```C#
services.AddLoggingConsole();
```

Reply:
Targetting IHttpClientBuilder implies that the functionality is capable of being isolated to the HttpClient that a particular instance of the builder produces. That isn't how the latency telemetry was originally designed to work. So the options are:
1. Ship the feature as it exists now on IServiceCollection
2. Redesign the feature so that it supports per-HttpClient isolation and then target IHttpClientBuilder
3. Don't ship the feature at all

The current plan is (1), you'd have to negotiate with Martin if you want to see how viable (2) is, and of course if you feel strongly enough you can ask for (3) instead.